### ISSUE TITLE
Allow the system to choose a port number

### ISSUE THREAD

**Issue by karwa** [swhitty/FlyingFox#41]
I'd like to use this server in some tests. The tests might be executed in parallel, so I'm not able to hard-code specific port numbers. 

If you `bind` a socket address with port number 0, the system will assign a unique port number for you, which is ideal for these situations. Unfortunately, AFAICT, there is no way to get the port number or socket from `HTTPServer`, so I wouldn't be able to actually make requests to that server instance.

**Comment by swhitty** [comment]
Great suggestion @karwa — the code to do this was [really simple](https://github.com/swhitty/FlyingFox/pull/42) and has been merged to the mainline branch and will be included in the next release.

I have updated a few of the [HTTPServerTests](https://github.com/swhitty/FlyingFox/blob/65b1f714edbe858505ad94963cf8af2e5f29c396/FlyingFox/Tests/HTTPServerTests.swift#L392) to use this technique and it is working well.

`HTTPServer` now includes a property that returns an optional `Socket.Address` indicating the address of the bounds listening socket.  You can then write a simple extension in your unit tests to wait for the server to start listening and then return the port like so;

```swift
extension HTTPServer {

  func waitForListeningPort() async throws -> UInt16 {
    try await waitUntilListening()
     switch listeningAddress {
     case let .ip4(_, port: port),
          let .ip6(_, port: port):
        return port
     default:
       throw Error.noPort
      }
  }

}
```

**Comment by karwa** [comment]
That’s great, thanks!

**Comment by swhitty** [comment]
Included in [0.9.0](https://github.com/swhitty/FlyingFox/releases/tag/0.9.0) release
